target/linux/ramips/dts/WNPR2600G.dts

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "iodata,wnpr2600g", "mediatek,mt7621-soc";
+	model = "I-O DATA WNPR2600G";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "wnpr2600g:green:power";
+			gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		notification {
+			label = "wnpr2600g:green:notification";
+			gpios = <&gpio0 12 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan2g {
+			label = "wnpr2600g:green:wlan2g";
+			gpios = <&gpio0 14 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan5g {
+			label = "wnpr2600g:green:wlan5g";
+			gpios = <&gpio0 15 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		auto {
+			label = "auto";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+			linux,input-type = <EV_SW>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
10MHz is a pretty slow SPI frequency. You could try if a higher frequency works for you.
MTK uses 30MHz in their SDK. All my mt7621/mt7628 boards works fine with 40MHz.

Reply:
Sounds great, I'll try to use higher clock.

Reply:
Done, 40MHz works and reading from flash has become faster. (writing has not been tested)

```
# 10MHz
root@OpenWrt:/tmp/mtd# time dd if=/dev/mtdblock3 of=mtd3_firmware.bin
27904+0 records in
27904+0 records out
real    0m 12.92s
user    0m 0.03s
sys     0m 0.38s

# 40MHz
root@OpenWrt:/tmp/mtd# time dd if=/dev/mtdblock3 of=mtd3_firmware.bin
27904+0 records in
27904+0 records out
real    0m 4.40s
user    0m 0.01s
sys     0m 0.56s
```